### DISCUSSION TITLE
podman - Storage for container has been removed

### DISCUSSION THREAD

**Discussion by baopipi0707** [containers/podman#17667] [Q&A]
I'm having issues using Podman with private repositories. I set TMPDIR as /.export/log/28890/.
When I use Podman to run images from a private registry I receive the error:

time="2023-02-28T14:50:46+08:00" level=error msg="Storage for container 01e52256df957963ced396582aecbac2ddd0d901bb9d5f9d13c4186deeb1548b has been removed"
Error: error writing runtime spec JSON for container 01e52256df957963ced396582aecbac2ddd0d901bb9d5f9d13c4186deeb1548b to disk: open /.export/log/28890/1253/share/containers/storage/overlay-containers/01e52256df957963ced396582aecbac2ddd0d901bb9d5f9d13c4186deeb1548b/userdata/config.json: no such file or directory.

**Comment by rhatdan** [comment]
Any chance you could generate this failure without podman compose?  Podman-compose isa different project and maintainers of Podman do not provide support for it.

**Reply by baopipi0707** [reply]
Sorry, my type error. I used podman not podman compose.

**Reply by rhatdan** [reply]
What command are you executing when you get this error?
Also supply `podman info`

**Reply by baopipi0707** [reply]
Thanks for the reply.
I excuted command: mpirun -np 32 --hostfile mfile podman --log-level=debug run --rm --env-host --env-file envfile --userns=keep-id --network=host --pid=host --ipc=host --privileged -v /tmp/podman-mpirun-1253:/tmp/podman-mpirun-1253:z -v ./workspace:/workspace:z -w /workspace abacus:3.1.0 ./abacus.sh to run podman.

Outputs of podman info is:
host:
  arch: amd64
  buildahVersion: 1.24.6
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /export/home/0248/.local/container/bin/conmon
    version: 'conmon version 2.1.0, commit: unknown'
  cpus: 64
  distribution:
    distribution: '"centos"'
    version: "7"
  eventLogger: file
  hostname: f0709
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1254
      size: 1
    - container_id: 1
      host_id: 9055072
      size: 35536
    uidmap:
    - container_id: 0
      host_id: 1253
      size: 1
    - container_id: 1
      host_id: 9055072
      size: 35536
  kernel: 3.10.0-1160.25.1.el7.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 512300802048
  memTotal: 540300972032
  networkBackend: cni
  ociRuntime:
    name: runc
    package: Unknown
    path: /export/home/0248/.local/container/sbin/runc
    version: |-
      runc version 1.1.0
      spec: 1.0.2-dev
      go: go1.16.13
      libseccomp: 2.3.1
  os: linux
  remoteSocket:
    path:/.export/log/28890/podman-run-1253/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_AUDIT_WRITE,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_MKNOD,CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: ""
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /export/home/0248/.local/container/bin/slirp4netns
    package: Unknown
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1.55-888d
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.3.1
  swapFree: 0
  swapTotal: 0
  uptime: 25h 24m 42.38s (Approximately 1.04 days)
plugins:
  log:
  - k8s-file
  - none
  - none
  - passthrough
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries: {}
store:
  configFile: /export/home/0248/.local/container/config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.imagestore: /export/home/0248/.local/share/containers/storage
    overlay.mount_program:
      Executable: /export/home/0248/.local/container/bin/fuse-overlayfs
      Package: Unknown
      Version: |-
        fusermount3 version: 3.10.4
        fuse-overlayfs: version 1.8.2
        FUSE library version 3.10.3
        using FUSE kernel interface version 7.31
    overlay.mountopt: nodev,fsync=0
    overlay.skip_mount_home: "true"
  graphRoot:/.export/log/28890/1253/share/containers/storage
  graphStatus:
    Backing Filesystem: tmpfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir:/.export/log/28890
  imageStore:
    number: 4
  runRoot: /.export/log/28890/containers-user-1253/containers
  volumePath:/.export/log/28890/1253/share/containers/storage/volumes
version:
  APIVersion: 4.0.2
  Built: 1677210116
  BuiltTime: Fri Feb 24 11:41:56 2023
  GitCommit: d54ecc8c3b635ee79d60a6eb4b04bd2a046f60f0-dirty
  GoVersion: go1.17.7
  OsArch: linux/amd64
  Version: 4.0.2
